### PR TITLE
feat(mapper): add LatexMergeTexMapper to extract and merge .tex files from archives

### PR DESCRIPTION
**Pull request by leeyyi**
## Summary

- Add a new **LatexMergeTexMapper** operator that extracts all `.tex` files from compressed LaTeX project archives (`.tar`, `.tar.gz`, `.tgz`, `.zip`) and concatenates them into a single text field.
- Includes zip-bomb protection via configurable `max_file_size`, graceful handling of non-UTF-8 encodings (Latin-1 bytes replaced with U+FFFD), and support for custom separators and field keys.
- Comprehensive test coverage with 14 unit tests covering multiple archive formats, encoding edge cases, size limits, and error handling.

## Changes

- `data_juicer/ops/mapper/latex_merge_tex_mapper.py` — New operator implementation
- `data_juicer/ops/mapper/__init__.py` — Register the new operator
- `data_juicer/config/config_all.yaml` — Add default config entry
- `docs/operators/mapper/latex_merge_tex_mapper.md` — Operator documentation (EN/CN)
- `tests/ops/mapper/test_latex_merge_tex_mapper.py` — 14 unit tests

## Test plan

- [x] All 14 unit tests pass locally

### CODE REVIEW

File: data_juicer/ops/mapper/latex_merge_tex_mapper.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                info = zf.getinfo(name)
+                if max_file_size and info.file_size > max_file_size:
+                    logger.warning(
+                        f"Skipping {name} in {archive_path}: "
+                        f"declared size {info.file_size} bytes exceeds "
+                        f"limit of {max_file_size} bytes")
+                    continue
+                contents.append(
+                    zf.read(name).decode("utf-8", errors="replace"))
+        return contents
```
[the remark this answers is not in the document]
Comment (leeyyi) [reply data_juicer/ops/mapper/latex_merge_tex_mapper.py]: Thanks for the suggestion. The performance difference here is negligible for our use case (LaTeX archives typically contain very few .tex files), and directory entries won't match the `.tex` suffix filter. I'll leave this as-is for now to keep the code straightforward.

File: data_juicer/ops/mapper/latex_merge_tex_mapper.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            logger.exception(
+                f"Failed to read archive {archive_path}")
+            return []
+
+    @staticmethod
+    def _read_tar(archive_path: str, max_file_size: int = 0):
+        contents = []
+        with tarfile.open(archive_path, 'r:*') as tf:
+            for member in tf.getmembers():
+                if not member.isfile():
```
[the remark this answers is not in the document]
Comment (leeyyi) [reply data_juicer/ops/mapper/latex_merge_tex_mapper.py:90]: Fixed in the latest commit.

File: data_juicer/ops/mapper/latex_merge_tex_mapper.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    def process_single(self, sample):
+        if self.compressed_file_key not in sample:
+            raise ValueError(
+                f"Compressed file key '{self.compressed_file_key}' "
+                f"not found in sample. "
+                f"Available keys: {list(sample.keys())}")
+
+        path = sample[self.compressed_file_key]
+        tex_contents = self._extract_tex_contents(path)
+        sample[self.text_key] = self.separator.join(tex_contents)
```
[the remark this answers is not in the document]
Comment (leeyyi) [reply data_juicer/ops/mapper/latex_merge_tex_mapper.py:184]: Fixed in the latest commit.
